### ISSUE TITLE
Errors are swallowed in selectors

### ISSUE THREAD

**Issue by ogix** [ngxs/store#843]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[ ] Other... Please describe:
</code></pre>

## Current behavior

When error occurs in a selector - it is swallowed and not shown in the console.

## Expected behavior

Error should be shown in the console.

## Minimal reproduction of the problem with instructions

```
  @Selector() 
  static pandas(state: string[]) {
    const foo: any = null;
    console.log(foo.bar);
    return state.filter(s => s.indexOf('panda') > -1);
  }
```

## What is the motivation / use case for changing the behavior?

Finding bugs easier in selectors.

## Environment

<pre><code>
Libs:
- @angular/core version: 7.1.4
- @ngxs/store version: 3.3.3

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by arturovt** [comment]
@ogix selectors calls are wrapped in `try-catch`, this decision was made in order to avoid exceptions. E.g. you've got some state that stores a `null` value or some object, and you want to create a selector that returns `token` value

```TS
@State<Token | null>({
  name: 'token',
  defaults: null
})
class TokenState {
  @Selector()
  public static getAccessToken(state: Token) {
    return state.accessToken;
  }
}
```
this won't throw and will return `undefined`.

@splincode what do you think?

**Comment by eranshmil** [comment]
@arturovt A `console.log` could be added to the `catch`, if `developmentMode` is turned on.

**Comment by dominicglenn** [comment]
`console.error` perhaps instead when dev mode?

**Comment by ogix** [comment]
Isn't it possible to check if state is null before applying a selector function? And then replace try/catch with that check.

**Comment by sroettering** [comment]
The swallowing is pretty annoying for testing, too. Most of the time I have to guess which properties I forgot to add to the initial state setup.

I think at least logging in dev mode should be added.

**Comment by splincode** [comment]
@ogix @sroettering You can try a new options.

Will be released 3.5.0

```ts
$ npm install @ngxs/store@dev
```

```ts
@NgModule({
  imports: [
       NgxsModule.forRoot([TasksState], {
            developmentMode: true,
            selectorOptions: { suppressErrors: false } // default: true
       })
  ]
})
export class AppModule {}
```

**Comment by ogix** [comment]
Thanks!

**Comment by splincode** [comment]
Released in 3.5.0

**Comment by jbjhjm** [comment]
Sorry for dedusting but I really wonder why this decision was made.
Defaulting to hiding errors produced by buggy code is pretty awful.
Errors indicate that something is going wrong. Without them, no chance notice something is wrong.
Why would anyone want to hide them? Only when visible devs can react and fix it.
